readthedocs/builds/querysets.py

```diff
@@ -118,6 +118,28 @@ def public(
     def api(self, user=None):
         return self.public(user, only_active=False)
 
+    def for_reindex(self):
+        """
+        Get all versions that can be reindexed.
+
+        A version can be reindexed if:
+
+        - It's active and has been built at least once successfully.
+          Since that means that it has files to be indexed.
+        - Its project is not delisted or marked as spam.
+        """
+        return (
+            self.filter(
+                active=True,
+                built=True,
+                builds__state=BUILD_STATE_FINISHED,
+                builds__success=True,
+            )
+            .exclude(project__delisted=True)
+            .exclude(project__is_spam=True)
```

Comment:
We should probably use a score here. Otherwise, only projects manually marked as spam will be excluded here.

Reply:
This is a copy of https://github.com/readthedocs/readthedocs.org/blob/53e21bb3ee8a5fce0194eb5481fd81ac87d3d1fa/readthedocs/search/documents.py#L120-L130

probably better discuss this at https://github.com/readthedocs/readthedocs.org/issues/9899